### PR TITLE
JVM: support inject fault to MySQL Java client

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
1. close issue #104 
2. update stress and gc in JVM, use https://github.com/chaos-mesh/byteman-helper

## Usage

```shell
chaosd attack jvm mysql --help
inject fault into MySQL client

Usage:
  chaosd attack jvm mysql [options] [flags]

Flags:
  -d, --database string               the match database
      --exception string              the exception message needs to throw
  -h, --help                          help for mysql
      --latency int                   the latency duration, unit ms
  -v, --mysql-connector-version int   the version of mysql-connector-java, only support 5.X.X(set to 5) and 8.X.X(set to 8) (default 5)
  -t, --table string                  the match table

Global Flags:
      --log-level string   the log level of chaosd. The value can be 'debug', 'info', 'warn' and 'error'
      --pid int            the pid of Java process which need to attach
      --port int           the port of agent server (default 9288)
      --uid string         the experiment ID
```

### CODE REVIEW

**Review by Andrewmatilde — APPROVED** [review]
LGTM.

**Comment by Andrewmatilde** [comment]
PTAL @cwen0

**Comment by Andrewmatilde** [comment]
I think we should add integration test for JVM test.

File: pkg/server/chaosd/jvm.go [hunk 1]
```diff
@@ -83,13 +60,25 @@ func (j jvmAttack) Attack(options core.AttackConfig, env Environment) (err error
 		log.Debug(string(output), zap.Error(err))
 	}
 
+	// submit helper jar
+	bmSubmitCmd := fmt.Sprintf(bmSubmitCommand, attack.Port, "b", fmt.Sprintf("%s/lib/byteman-helper.jar", os.Getenv("BYTEMAN_HOME")))
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply pkg/server/chaosd/jvm.go:64]: why need to do this, I think it's the internal implementation logic

**Comment by WangXiangUSTC** [comment]
> I think we should add e2e-test for JVM test.

You are right, I add a integration test, PTAL again @Andrewmatilde

**Review by Andrewmatilde — APPROVED** [review]
LGTM

**Comment by cwen0** [comment]
@WangXiangUSTC We need a new document to guide users on how to use it.

**Comment by WangXiangUSTC** [comment]
> @WangXiangUSTC We need a new document to guide users on how to use it.

https://github.com/chaos-mesh/website/issues/245 create an issue for it @cwen0

**Comment by cwen0** [comment]
@WangXiangUSTC How about split to two pr? You can update stress and gc in JVM by using https://github.com/chaos-mesh/byteman-helper in a single pr.

**Comment by WangXiangUSTC** [comment]
> @WangXiangUSTC How about split to two pr? You can update stress and gc in JVM by using https://github.com/chaos-mesh/byteman-helper in a single pr.

OK

File: cmd/attack/jvm.go [hunk 2]
```diff
@@ -150,6 +151,26 @@ func NewJVMRuleFileCommand(dep fx.Option, options *core.JVMCommand) *cobra.Comma
...
+		Use:   "mysql [options]",
+		Short: "inject fault into MySQL client",
+		Run: func(*cobra.Command, []string) {
+			options.Action = core.JVMMySQLAction
+			utils.FxNewAppWithoutLog(dep, fx.Invoke(jvmCommandFunc)).Run()
+		},
+	}
+
+	cmd.Flags().StringVarP(&options.SQLType, "sql-type", "", "", "the match sql type")
+	cmd.Flags().StringVarP(&options.Database, "database", "d", "", "the match database")
```
Comment (cwen0) [inline cmd/attack/jvm.go:165]: If this field is required, we can add the required information on help descriptipn.

Comment (WangXiangUSTC) [reply cmd/attack/jvm.go:165]: ditto, I will update the description and comment

File: pkg/core/jvm.go [hunk 3]
```diff
@@ -123,6 +137,24 @@ type JVMStressSpec struct {
 	MemoryType string `json:"mem-type,omitempty"`
 }
 
+// only when SQL match the Database, Table, SQLType, chaosd will inject fault
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply pkg/core/jvm.go:140]: It passed the check with the Grammarly 🥲

**Comment by WangXiangUSTC** [comment]
I updated the comment, PTAL again @cwen0

File: pkg/core/jvm.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cwen0) [reply pkg/core/jvm.go:140]: It is customary for go comments to begin with the "JVMMySQLSpec" and introduce the general information about this strcut.

[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply pkg/core/jvm.go]: done

**Review by cwen0 — APPROVED** [review]
LGTM

**Pull request merged**